rocketpy/mathutils/function.py

```diff
@@ -1225,13 +1297,14 @@ def plot2D(
         x = np.linspace(lower[0], upper[0], samples[0])
         y = np.linspace(lower[1], upper[1], samples[1])
         mesh_x, mesh_y = np.meshgrid(x, y)
-        mesh_x_flat, mesh_y_flat = mesh_x.flatten(), mesh_y.flatten()
-        mesh = [[mesh_x_flat[i], mesh_y_flat[i]] for i in range(len(mesh_x_flat))]
+        mesh = np.column_stack((mesh_x.flatten(), mesh_y.flatten()))
+
         # Evaluate function at all mesh nodes and convert it to matrix
-        z = np.array(self.get_value(mesh)).reshape(mesh_x.shape)
+        z = np.array(self.get_value(mesh[:, 0], mesh[:, 1])).reshape(mesh_x.shape)
```

Comment:
This line is causing a mistake due the inner workings of shepard interpolation do not expect the data in this format. What I noticed:

1. Shepard iterator check needs to be updated to handle ndarrays. This is easy I may be done with `collections.abc.Iterable` or adding `np.ndarray` to the list of `isinstance`. However this alone does not fix the mistake.
2. The `x` data from shepard expects a stacked format instead of the first argument list. Still working on a solve.

Just so you can compare:

Before

![image](https://github.com/RocketPy-Team/RocketPy/assets/87212571/dcfb1cab-8778-4f9d-9b43-1a5845d12844)


After

![image](https://github.com/RocketPy-Team/RocketPy/assets/87212571/f5bef5dd-1bca-4720-88b4-7576e0c0c71f)

Reply:
Solved by [0f76b3e](https://github.com/RocketPy-Team/RocketPy/pull/451/commits/0f76b3e4c4a9c35155a6543e6d59afae6e7ce7b3).